### DISCUSSION TITLE
Need help with exporting spans using JaegerExporter from python. I keep getting Bad version in ReadMessageBegin

### DISCUSSION THREAD

**Discussion by vamshiaruru-virgodesigns** [jaegertracing/jaeger#3673] [Q&A]
Hi all,

I am trying to push spans from python fastapi app to jaeger. I brought up jaeger locally using the `all-in-one` image, using this command:
```
docker run -d --name jaeger \
  -e COLLECTOR_ZIPKIN_HTTP_PORT=9411 \
  -p 5775:5775/udp \
  -p 6831:6831/udp \
  -p 6832:6832/udp \
  -p 5778:5778 \
  -p 16686:16686 \
  -p 14268:14268 \
  -p 9411:9411 \
  jaegertracing/all-in-one:latest
  ```
  
  And in python, I am setting up open telemetry like so:
  ```
      trace.set_tracer_provider(
        TracerProvider(resource=Resource.create({SERVICE_NAME: "app"}))
    )
    jaeger_exporter = JaegerExporter(
        agent_host_name="localhost",
        agent_port=6832,
    )
    trace.get_tracer_provider().add_span_processor(BatchSpanProcessor(jaeger_exporter))
```
So as traces are generated and pushed, I see the following error message in `jaeger` container logs:

```
{"level":"error","ts":1652354927.6273549,"caller":"processors/thrift_processor.go:123","msg":"Processor failed","error":"Bad version in ReadMessageBegin","stacktrace":"github.com/jaegertracing/jaeger/cmd/agent/app/processors.(*ThriftProcessor).processBuffer\n\tgithub.com/jaegertracing/jaeger/cmd/agent/app/processors/thrift_processor.go:123\ngithub.com/jaegertracing/jaeger/cmd/agent/app/processors.NewThriftProcessor.func2\n\tgithub.com/jaegertracing/jaeger/cmd/agent/app/processors/thrift_processor.go:87"}
```

I am able to push data to this jaeger instance using Node.JS exporter, but not python exporter. 

I have no idea what I am doing wrong, or how even to debug this. Any help is very appreciated.  Thanks :D

**Comment by vamshiaruru-virgodesigns** [comment]
It started working fine after changing port from 6832 to 6831. But it is working fine on JS with 6832. Is it because multiple exporters can't send to the same port?

**Comment by yurishkuro** [comment]
Ports 6831 and 6832 expect different Thrift encodings, binary vs. compact.

**Reply by vamshiaruru-virgodesigns** [reply]
Thanks, I did not know that. So is it the case that the python exporter does not support compact encoding but javascript exporter does?

**Reply by yurishkuro** [reply]
All languages support compact encoding except for Javascript, that one only had binary encoding, so we had to introduce a new port for it. We actually have an open ticket to automatically sense the different encoding coming to a port so that a single port could be used.

**Reply by yurishkuro** [reply]
https://github.com/jaegertracing/jaeger/issues/1596

**Reply by vamshiaruru-virgodesigns** [reply]
okay, TIL. Thank you :D
